### PR TITLE
Updated Nunavut to 2.1.1

### PR DESCRIPTION
**Pull request by freshnikolas**

### CODE REVIEW

File: nunaserver/docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: "3"
 services:
   celery:
-    image: python:latest
+    image: python:3.9
```
Comment (pavel-kirienko) [inline nunaserver/docker-compose.yml]: Remind me please, why is it not possible to use the latest Python here?

File: nunaserver/requirements.txt [hunk 2]
```diff
@@ -9,11 +9,11 @@ requests==2.25.1
...
 # Celery
 celery==5.1.0
-redis==3.5.3
+redis==4.4.4
 
 # Nunavut
-nunavut==1.3.0
-pydsdl==1.12.1
+nunavut==2.1.1
+pydsdl==1.18
```
Comment (pavel-kirienko) [inline nunaserver/requirements.txt]:
```suggestion
```

PyDSDL of the correct version will be pulled in through Nunavut.

File: nunaserver/requirements.txt [hunk 3]
```diff
@@ -9,11 +9,11 @@ requests==2.25.1
 
 # Celery
 celery==5.1.0
-redis==3.5.3
+redis==4.4.4
 
 # Nunavut
-nunavut==1.3.0
-pydsdl==1.12.1
+nunavut==2.1.1
```
Comment (pavel-kirienko) [inline nunaserver/requirements.txt]:
```suggestion
nunavut~=2.1.1
```

**Review by pavel-kirienko — COMMENTED** [review]
Please add a project config file named `pyproject.toml` and add a section for Black there taken from here: https://github.com/Zubax/epm_hardware/blob/ee0493d96ff5dbe3cbd41522f507b50bf58cdb83/force_measurement_rig/client/pyproject.toml#L87-L95. This is needed to format the code correctly with the maximum line length of 120 characters.

File: nunaserver/docker-compose.yml (hunk 1 above)
Comment (freshnikolas) [reply nunaserver/docker-compose.yml]: Because Celery 5.1 is only supported up to Python 3.9.

Comment (pavel-kirienko) [reply nunaserver/docker-compose.yml]: Can we switch to the latest celery?

File: nunaserver/requirements.txt (hunk 2 above)
Comment (freshnikolas) [reply nunaserver/requirements.txt]: should I delete this line completely or just revert the changes?

Comment (pavel-kirienko) [reply nunaserver/requirements.txt]: Please delete it

**Comment by freshnikolas** [comment]
> Please add a project config file named `pyproject.toml` and add a section for Black there taken from here: https://github.com/Zubax/epm_hardware/blob/ee0493d96ff5dbe3cbd41522f507b50bf58cdb83/force_measurement_rig/client/pyproject.toml#L87-L95. This is needed to format the code correctly with the maximum line length of 120 characters.

I don"t think I have access to this repo. I get a 404 error.

**Comment by pavel-kirienko** [comment]
You should have access now.

File: nunaserver/docker-compose.yml (hunk 1 above)
Comment (freshnikolas) [reply nunaserver/docker-compose.yml]: will test it

Comment (freshnikolas) [reply nunaserver/docker-compose.yml]: works with the newest version of Celery

File: nunaserver/docker-compose.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ services:
       - NS_LOG_LEVEL=20
     command: "bash -c 'cd /nunaserver && pip3 install -r requirements.txt && celery -A nunaserver.generator worker --loglevel=info'"
   nunaserver:
-    image: python:latest
+    image: python:3.9
```
Comment (pavel-kirienko) [inline nunaserver/docker-compose.yml]: Same question here. Can we lift this restriction? If not, why?

File: nunaserver/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# --------------------------------------------------  BLACK  --------------------------------------------------
+[tool.black]
+line-length = 120
+target-version = ['py311']
+include = '''
+(src/.*\.pyi?$)
```
Comment (pavel-kirienko) [inline nunaserver/pyproject.toml]:
```suggestion
(nunaserver/.*\.pyi?$)
```

Please reformat the code afterward using Black.

**Review by pavel-kirienko — CHANGES_REQUESTED** [review]

File: nunaserver/requirements.txt [hunk 6]
```diff
@@ -8,13 +8,12 @@ gunicorn==20.1.0
 requests==2.25.1
 
 # Celery
-celery==5.1.0
-redis==3.5.3
+celery==5.3.1
+redis==4.4.4
```
Comment (pavel-kirienko) [inline nunaserver/requirements.txt]: The latest one is 4.6.0, maybe we should use that?

File: nunaserver/docker-compose.yml (hunk 4 above)
Comment (freshnikolas) [reply nunaserver/docker-compose.yml]: The restriction has already been lifted.
This is an old change. In the last push I changed it back to "python:latest"

Comment (pavel-kirienko) [reply nunaserver/docker-compose.yml]: It is not yet changed. Maybe you forgot to push the latest change.

Comment (freshnikolas) [reply nunaserver/docker-compose.yml]: Whoopsie daisy, you are right...
Done!

**Review by pavel-kirienko — COMMENTED** [review]
Nice. Are there any docs or READMEs that require updating? Please check these and see if there's anything made obsolete by your changes.

Is it going to be deployed to production automatically? I suppose it is as we have a workflow for that, please monitor how it performs.

**Review by pavel-kirienko — APPROVED** [review]

**Pull request merged**